firmware/controllers/trigger/trigger_central.cpp

```diff
@@ -210,42 +210,37 @@ static angle_t wrapVvt(angle_t vvtPosition, int period) {
 	return vvtPosition;
 }
 
-static void logFront(bool isImportantFront, efitick_t nowNt, int index) {
+static void logVvtFront(bool useOnlyRise, bool isImportantFront, TriggerValue front, efitick_t nowNt, int index) {
 	if (isImportantFront && isBrainPinValid(engineConfiguration->camInputsDebug[index])) {
 #if EFI_PROD_CODE
 		writePad("cam debug", engineConfiguration->camInputsDebug[index], 1);
 #endif /* EFI_PROD_CODE */
 		getExecutorInterface()->scheduleByTimestampNt("dbg_on", &debugToggleScheduling, nowNt + DEBUG_PIN_DELAY, &turnOffAllDebugFields);
 	}
 
-	if (engineConfiguration->displayLogicLevelsInEngineSniffer && isImportantFront) {
-		if (engineConfiguration->vvtCamSensorUseRise) {
-			// todo: unify TS composite logger code with console Engine Sniffer
-			// todo: better API to reduce copy/paste?
+	if (!useOnlyRise || engineConfiguration->displayLogicLevelsInEngineSniffer) {
+		// If we care about both edges OR displayLogicLevel is set, log every front exactly as it is
+		addEngineSnifferVvtEvent(index, front == TriggerValue::RISE ? FrontDirection::UP : FrontDirection::DOWN);
+
 #if EFI_TOOTH_LOGGER
-			LogTriggerTooth(SHAFT_SECONDARY_RISING, nowNt);
-			LogTriggerTooth(SHAFT_SECONDARY_FALLING, nowNt);
+		LogTriggerTooth(front == TriggerValue::RISE ? SHAFT_SECONDARY_RISING : SHAFT_SECONDARY_FALLING, nowNt);
 #endif /* EFI_TOOTH_LOGGER */
+	} else {
+		if (isImportantFront) {
+			// On the important edge, log a rise+fall pair, and nothing on the real falling edge
 			addEngineSnifferVvtEvent(index, FrontDirection::UP);
 			addEngineSnifferVvtEvent(index, FrontDirection::DOWN);
-		} else {
+
 #if EFI_TOOTH_LOGGER
-			LogTriggerTooth(SHAFT_SECONDARY_FALLING, nowNt);
 			LogTriggerTooth(SHAFT_SECONDARY_RISING, nowNt);
+			LogTriggerTooth(SHAFT_SECONDARY_FALLING, nowNt);
 #endif /* EFI_TOOTH_LOGGER */
-
-			addEngineSnifferVvtEvent(index, FrontDirection::DOWN);
-			addEngineSnifferVvtEvent(index, FrontDirection::UP);
 		}
 	}
 }
 
 void hwHandleVvtCamSignal(TriggerValue front, efitick_t nowNt, int index) {
 	TriggerCentral *tc = getTriggerCentral();
-	if (tc->directSelfStimulation || !tc->hwTriggerInputEnabled) {
```

Comment:
A bit surprised to see these lines removed under "logging logic"?

Reply:
as you should be, that was for testing 🙂